pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+	for _, eip := range eipList.Items {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		if err := deleteCloudPrivateIpConfigs(ctx, client, egressIps); err != nil {
+			return err
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+		eipNamespaceStr := fmt.Sprintf("%v", eipNamespace)
+		nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespaceStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		nnsResourceVersion := nns.Object["metadata"].(map[string]interface{})["resourceVersion"]
+		nnsNetId := nns.Object["netid"]
+		netName := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+
+		netnamespace := &uns.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "network.openshift.io/v1",
+				"kind":       "NetNamespace",
+				"metadata": map[string]interface{}{
+					"name":            netName,
+					"resourceVersion": nnsResourceVersion,
+				},
+				"netid":     nnsNetId,
+				"netname":   netName,
+				"egressIPs": egressIps,
+			},
+		}
+		if err := apply.ApplyObject(ctx, client, netnamespace, ""); err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList.Items {
+		if _, ok := node.Labels["k8s.ovn.org/egress-assignable"]; ok {
+			// parse out node's subnet from egressip annotation
+			egressIpConfigParsed := strings.SplitAfter(node.Annotations["cloud.network.openshift.io/egress-ipconfig"], "\"ipv4\":\"")
+			nodeSubnet := egressIpConfigParsed[1]
+			nodeSubnet = nodeSubnet[:strings.IndexByte(nodeSubnet, '"')]
+			nodeSubnetArr := []string{nodeSubnet}
+
+			hostIpParsed := strings.SplitAfter(node.Annotations["k8s.ovn.org/host-addresses"], "[\"")
+			hostIp := hostIpParsed[1]
+			hostIp = hostIp[:strings.IndexByte(hostIp, '"')]
+
+			hsn, err := client.Default().Dynamic().Resource(gvrHostSubnet).Get(ctx, node.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			hsnResourceVersion := hsn.Object["metadata"].(map[string]interface{})["resourceVersion"]
+			hsnSubnet := hsn.Object["subnet"]
+
+			hostSubnet := &uns.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion":  "network.openshift.io/v1",
+					"kind":        "HostSubnet",
+					"egressCIDRs": nodeSubnetArr,
+					"host":        node.Name,
+					"hostIP":      hostIp,
+					"metadata": map[string]interface{}{
+						"name":            node.Name,
+						"resourceVersion": hsnResourceVersion,
+					},
+					"subnet": hsnSubnet,
```

Comment:
Should some of this stuff be inside a "spec"?
Also, if we are just updating egressCIDRs, why not do so directly in `hsnSubnet`?

Reply:
This was the product of old code sticking around. You are correct, I can just set them directly in `hsn`. (I assume you mean `hsn` instead of `hsnSubnet`).

Regarding the "spec" comment:
> No, see bindata/network/openshift-sdn/001-crd.yaml. It is a property, like apiVersion. I find it...strange as well. But it is what it is.